packages/shared/src/hooks/feed/useFeedUpvotePost.ts

```diff
@@ -27,24 +49,72 @@ export default function useFeedUpvotePost(
   column: number,
   upvoted: boolean,
 ) => Promise<void> {
+  const queryClient = useQueryClient();
   const { user, showLogin } = useContext(AuthContext);
   const { trackEvent } = useContext(AnalyticsContext);
 
-  const { upvotePost, cancelPostUpvote } = useUpvotePost<{
-    id: string;
-    index: number;
-  }>({
-    onUpvotePostMutate: optimisticPostUpdateInFeed(
-      items,
-      updatePost,
-      (post) => ({ upvoted: true, numUpvotes: post.numUpvotes + 1 }),
-    ),
-    onCancelPostUpvoteMutate: optimisticPostUpdateInFeed(
-      items,
-      updatePost,
-      (post) => ({ upvoted: false, numUpvotes: post.numUpvotes - 1 }),
-    ),
-  });
+  const { upvotePost, cancelPostUpvote } =
+    useUpvotePost<UseFeedUpvotePostVariables>({
+      onUpvotePostMutate: optimisticPostUpdateInFeed(
+        items,
+        updatePost,
+        mutationHandlers[upvotePostKey],
+      ),
+      onCancelPostUpvoteMutate: optimisticPostUpdateInFeed(
+        items,
+        updatePost,
+        mutationHandlers[cancelUpvotePostKey],
+      ),
+    });
+
+  useEffect(() => {
+    const matchedMutations = [
+      upvotePostMutationKey,
+      cancelUpvotePostMutationKey,
+    ].map((item) => item.toString());
+
+    const unsubscribe = queryClient.getMutationCache().subscribe((event) => {
+      if (event.state.status !== 'success') {
+        return;
+      }
+
+      const mutationKey = event.options.mutationKey?.toString();
+
+      if (matchedMutations.includes(mutationKey)) {
+        const { id, index: indexFromEvent } = event.options
+          .variables as unknown as UseFeedUpvotePostVariables;
+        const mutationFromFeedHook = typeof indexFromEvent !== 'undefined';
+
+        if (mutationFromFeedHook) {
+          return;
+        }
+
+        const index = items.findIndex(
+          (item) => item.type === 'post' && item.post.id === id,
+        );
+
+        if (index === -1) {
+          return;
+        }
+
+        const mutationHandler = mutationHandlers[mutationKey];
+
+        if (!mutationHandler) {
+          return;
+        }
```

Comment:
This could be outside the if even, as it would be early return right?

Reply:
Oh yeah this could actually replace `matchedMutations.includes(mutationKey)` on the top. 👍